No output from flake8 on Windows

### Discussed in https://github.com/akaihola/darker/discussions/368

<div type='discussions-op-text'>

<sup>Originally posted by **deadkex** May 20, 2022</sup>
Hi!
I am using darker in combination with pre-commit and the linters i would like to use are pylint and flake8.
The problem is that i cannot get any output from flake8, on the other hand pylint works as expected.
If you take below config and `pre-commit run --file test.py`, you can see that flake8 is actually running and reporting violations, but darker is not showing them. I also tried committing larger files with lots of changes or explicitly adding violations as changes.

python: 3.8.10
darker: 1.5.0
pre-commit: 2.19.0
flake8: 4.0.1
black: 22.3.0

.pre-commit-config.yaml
```
default_language_version:
    python: python3

repos:
-   repo: local
    hooks:
    -   id: darker
        name: darker
        entry: darker
        language: system
        types: [python]
        args: [
            "-L flake8 -v"
        ]
    -   id: flake8
        name: flake8
        entry: flake8
        language: system
        types: [ python ]
```

pyproject.toml
```
[tool.darker]
src = ["./"]
revision = ":PRE-COMMIT:"
diff = false
check = true
isort = false
# lint = [
#     "flake8"
# ]

[tool.black]
target-version = ["py38"]
include = '\.py?$'
```

test.py
```
try:
    pass
except:
    pass
```</div>

Using a file from my personal project

`darker -vv -L flake8 BDBot.py`
```
# Effective configuration:

[tool.darker]
src = [
    "BDBot.py",
]
revision = ":PRE-COMMIT:"
diff = false
stdout = false
check = true
isort = false
lint = [
    "flake8",
]
log_level = "DEBUG"
workers = 1


# Configuration options which differ from defaults:

[tool.darker]
src = [
    "BDBot.py",
]
revision = ":PRE-COMMIT:"
check = true
lint = [
    "flake8",
]
log_level = "DEBUG"



DEBUG:darker.git:[D:\Dokumente\Programming\python\_Bots\BDBot]$ git rev-parse --is-inside-work-tree
DEBUG:darker.git:[D:\Dokumente\Programming\python\_Bots\BDBot]$ git diff --name-only --relative HEAD -- BDBot.py
DEBUG:darker.git:[D:\Dokumente\Programming\python\_Bots\BDBot]$ git ls-files --others --exclude-standard -- BDBot.py
DEBUG:darker.__main__:Read 203 lines from edited file D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py
DEBUG:darker.__main__:Black reformat resulted in 280 lines
DEBUG:darker.diff:Diff between edited and reformatted has 41 opcodes
DEBUG:darker.git:[D:\Dokumente\Programming\python\_Bots\BDBot]$ git show HEAD:./BDBot.py
DEBUG:darker.git:[D:\Dokumente\Programming\python\_Bots\BDBot]$ git log -1 --format=%ct HEAD -- BDBot.py
DEBUG:darker.diff:Diff between edited and reformatted has 7 opcodes
DEBUG:darker.chooser:Found no edits on lines 1-6
DEBUG:darker.chooser:Using 6 original lines at line 1
DEBUG:darker.chooser:Found no edits on lines 7-10
DEBUG:darker.chooser:Using 4 original lines at line 7
DEBUG:darker.chooser:Found no edits on lines 11-14
DEBUG:darker.chooser:Using 4 original lines at line 11
DEBUG:darker.chooser:Found no edits on lines 15-16
DEBUG:darker.chooser:Using 2 original lines at line 15
DEBUG:darker.chooser:Found no edits on lines 17-51
DEBUG:darker.chooser:Using 35 original lines at line 17
DEBUG:darker.chooser:Found no edits on line 52
DEBUG:darker.chooser:Using 1 original line at line 52
DEBUG:darker.chooser:Found no edits on lines 53-56
DEBUG:darker.chooser:Using 4 original lines at line 53
DEBUG:darker.chooser:Found no edits on line 57
DEBUG:darker.chooser:Using 1 original line at line 57
DEBUG:darker.chooser:Found no edits on lines 58-59
DEBUG:darker.chooser:Using 2 original lines at line 58
DEBUG:darker.chooser:Found no edits on line 60
DEBUG:darker.chooser:Using 1 original line at line 60
DEBUG:darker.chooser:Found no edits on lines 61-62
DEBUG:darker.chooser:Using 2 original lines at line 61
DEBUG:darker.chooser:Found no edits on line 63
DEBUG:darker.chooser:Using 1 original line at line 63
DEBUG:darker.chooser:Found no edits on lines 64-65
DEBUG:darker.chooser:Using 2 original lines at line 64
DEBUG:darker.chooser:Found no edits on line 66
DEBUG:darker.chooser:Using 1 original line at line 66
DEBUG:darker.chooser:Found no edits on line 67
DEBUG:darker.chooser:Using 1 original line at line 67
DEBUG:darker.chooser:Found no edits on line 68
DEBUG:darker.chooser:Using 1 original line at line 68
DEBUG:darker.chooser:Found no edits on lines 69-76
DEBUG:darker.chooser:Using 8 original lines at line 69
DEBUG:darker.chooser:Found no edits on lines 77-79
DEBUG:darker.chooser:Using 3 original lines at line 77
DEBUG:darker.chooser:Found no edits on lines 80-96
DEBUG:darker.chooser:Using 17 original lines at line 80
DEBUG:darker.chooser:Found no edits on lines 97-103
DEBUG:darker.chooser:Using 7 original lines at line 97
DEBUG:darker.chooser:Found no edits on lines 104-106
DEBUG:darker.chooser:Using 3 original lines at line 104
DEBUG:darker.chooser:Found no edits on line 107
DEBUG:darker.chooser:Using 1 original line at line 107
DEBUG:darker.chooser:Found no edits on lines 108-112
DEBUG:darker.chooser:Using 5 original lines at line 108
DEBUG:darker.chooser:Found no edits on line 113
DEBUG:darker.chooser:Using 1 original line at line 113
DEBUG:darker.chooser:Found no edits on lines 114-119
DEBUG:darker.chooser:Using 6 original lines at line 114
DEBUG:darker.chooser:Found no edits on lines 120-121
DEBUG:darker.chooser:Using 2 original lines at line 120
DEBUG:darker.chooser:Found no edits on lines 122-135
DEBUG:darker.chooser:Using 14 original lines at line 122
DEBUG:darker.chooser:Found no edits on line 136
DEBUG:darker.chooser:Using 1 original line at line 136
DEBUG:darker.chooser:Found no edits on lines 137-157
DEBUG:darker.chooser:Using 21 original lines at line 137
DEBUG:darker.chooser:Found no edits on lines 158-159
DEBUG:darker.chooser:Using 2 original lines at line 158
DEBUG:darker.chooser:Found no edits on lines 160-165
DEBUG:darker.chooser:Using 6 original lines at line 160
DEBUG:darker.chooser:Found no edits on lines 166-167
DEBUG:darker.chooser:Using 2 original lines at line 166
DEBUG:darker.chooser:Found no edits on lines 168-173
DEBUG:darker.chooser:Using 6 original lines at line 168
DEBUG:darker.chooser:Found no edits on lines 174-177
DEBUG:darker.chooser:Using 4 original lines at line 174
DEBUG:darker.chooser:Found no edits on lines 178-181
DEBUG:darker.chooser:Using 4 original lines at line 178
DEBUG:darker.chooser:Found no edits on line 182
DEBUG:darker.chooser:Using 0 original lines at line 182
DEBUG:darker.chooser:Found edits on lines 182-188
DEBUG:darker.chooser:Using 7 unmodified lines at line 182
DEBUG:darker.chooser:Found edits on line 189
DEBUG:darker.chooser:Using 3 reformatted lines at line 189
DEBUG:darker.chooser:Found edits on lines 190-201
DEBUG:darker.chooser:Using 12 unmodified lines at line 190
DEBUG:darker.chooser:Found no edits on line 202
DEBUG:darker.chooser:Using 1 original line at line 202
DEBUG:darker.chooser:Found no edits on line 203
DEBUG:darker.chooser:Using 1 original line at line 203
DEBUG:darker.__main__:Verifying that the 203 original edited lines and 205 reformatted lines parse into an identical abstract syntax tree
DEBUG:darker.linting:[D:\Dokumente\Programming\python\_Bots\BDBot]$ flake8 D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:7:80: E501 line too long (112 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:9:80: E501 line too long (112 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:15:80: E501 line too long (105 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:30:80: E501 line too long (80 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:50:80: E501 line too long (87 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:51:80: E501 line too long (89 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:52:80: E501 line too long (96 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:57:80: E501 line too long (111 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:60:80: E501 line too long (94 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:63:80: E501 line too long (105 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:66:80: E501 line too long (98 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:68:80: E501 line too long (104 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:77:80: E501 line too long (108 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:78:80: E501 line too long (112 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:100:51: F541 f-string is missing placeholders
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:101:78: F541 f-string is missing placeholders
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:101:80: E501 line too long (83 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:107:80: E501 line too long (119 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:113:80: E501 line too long (109 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:120:80: E501 line too long (80 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:126:80: E501 line too long (84 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:136:80: E501 line too long (99 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:158:80: E501 line too long (88 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:159:80: E501 line too long (107 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:166:80: E501 line too long (88 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:167:80: E501 line too long (107 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:174:80: E501 line too long (82 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:175:80: E501 line too long (91 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:189:80: E501 line too long (133 > 79 characters)
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:195:1: E722 do not use bare 'except'
DEBUG:darker.linting:Unparsable linter output: D:\Dokumente\Programming\python\_Bots\BDBot\BDBot.py:202:80: E501 line too long (89 > 79 characters)
```

`flake8 BDBot.py`
```
D:\Dokumente\Programming\python\_Bots\BDBot>flake8 BDBot.py
BDBot.py:7:80: E501 line too long (112 > 79 characters)
BDBot.py:9:80: E501 line too long (112 > 79 characters)
BDBot.py:15:80: E501 line too long (105 > 79 characters)
BDBot.py:30:80: E501 line too long (80 > 79 characters)
BDBot.py:50:80: E501 line too long (87 > 79 characters)
BDBot.py:51:80: E501 line too long (89 > 79 characters)
BDBot.py:52:80: E501 line too long (96 > 79 characters)
BDBot.py:57:80: E501 line too long (111 > 79 characters)
BDBot.py:60:80: E501 line too long (94 > 79 characters)
BDBot.py:63:80: E501 line too long (105 > 79 characters)
BDBot.py:66:80: E501 line too long (98 > 79 characters)
BDBot.py:68:80: E501 line too long (104 > 79 characters)
BDBot.py:77:80: E501 line too long (108 > 79 characters)
BDBot.py:78:80: E501 line too long (112 > 79 characters)
BDBot.py:100:51: F541 f-string is missing placeholders
BDBot.py:101:78: F541 f-string is missing placeholders
BDBot.py:101:80: E501 line too long (83 > 79 characters)
BDBot.py:107:80: E501 line too long (119 > 79 characters)
BDBot.py:113:80: E501 line too long (109 > 79 characters)
BDBot.py:120:80: E501 line too long (80 > 79 characters)
BDBot.py:126:80: E501 line too long (84 > 79 characters)
BDBot.py:136:80: E501 line too long (99 > 79 characters)
BDBot.py:158:80: E501 line too long (88 > 79 characters)
BDBot.py:159:80: E501 line too long (107 > 79 characters)
BDBot.py:166:80: E501 line too long (88 > 79 characters)
BDBot.py:167:80: E501 line too long (107 > 79 characters)
BDBot.py:174:80: E501 line too long (82 > 79 characters)
BDBot.py:175:80: E501 line too long (91 > 79 characters)
BDBot.py:189:80: E501 line too long (133 > 79 characters)
BDBot.py:195:1: E722 do not use bare 'except'
BDBot.py:202:80: E501 line too long (89 > 79 characters)
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -16,12 +16,13 @@ Fixed
 -----
 - ``darker --revision=a..b .`` now works since the repository root is now always
   considered to have existed in all historical commits.
-- Ignore linter lines which refer to files outside the common root of paths on the
-  command line. Fixes a failure when Pylint notifies about obsolete options in
-  ``.pylintrc``.
+- Ignore linter lines which refer to non-Python files or files outside the common root
+  of paths on the command line. Fixes a failure when Pylint notifies about obsolete
+  options in ``.pylintrc``.
 - For linting Darker's own code base, require Pylint 2.6.0 or newer. This avoids the
   need to skip the obsolete ``bad-continuation`` check now removed from Pylint.
 - Fix linter output parsing for full Windows paths which include a drive letter.
+- Stricter rules for linter output parsing.
 
 
 1.5.0_ - 2022-04-23

--- a/src/darker/linting.py
+++ b/src/darker/linting.py
@@ -106,8 +106,12 @@ def _parse_linter_line(line: str, root: Path) -> Tuple[Path, int, str, str]:
     path_from_cwd = Path(path_str).absolute()
     try:
         path_in_repo = path_from_cwd.relative_to(root)
-    except ValueError as exc_info:
-        logger.debug("Unparsable linter output: %s (%s)", line[:-1], exc_info)
+    except ValueError:
+        logger.warning(
+            "Linter message for a file %s outside requested directory %s",
+            path_from_cwd,
+            root,
+        )
         return Path(), 0, "", ""
     return path_in_repo, linenum, location + ":", description
 
@@ -184,6 +188,13 @@ def run_linter(
                 or path_in_repo in missing_files
                 or linter_error_linenum == 0
             ):
+                continue
+            if path_in_repo.suffix != ".py":
+                logger.warning(
+                    "Linter message for a non-Python file: %s %s",
+                    location,
+                    description,
+                )
                 continue
             try:
                 edited_linenums = edited_linenums_differ.compare_revisions(

--- a/src/darker/linting.py
+++ b/src/darker/linting.py
@@ -39,7 +39,13 @@ def _parse_linter_line(line: str, root: Path) -> Tuple[Path, int, str, str]:
     #       description = "error: Foo"
     try:
         location, description = line[:-1].split(": ", 1)
-        path_str, linenum_str, *rest = location.split(":")
+        if location[1:3] == ":\\":
+            # Absolute Windows paths need special handling. Separate out the ``C:`` (or
+            # similar), then split by colons, and finally re-insert the ``C:``.
+            path_in_drive, linenum_str, *rest = location[2:].split(":")
+            path_str = f"{location[:2]}{path_in_drive}"
+        else:
+            path_str, linenum_str, *rest = location.split(":")
         linenum = int(linenum_str)
         if len(rest) > 1:
             raise ValueError("Too many colon-separated tokens")

--- a/src/darker/linting.py
+++ b/src/darker/linting.py
@@ -31,14 +31,56 @@ from darker.highlighting import colorize
 logger = logging.getLogger(__name__)
 
 
+def _strict_nonneg_int(text: str) -> int:
+    """Strict parsing of strings to non-negative integers
+
+    Allow no leading or trailing whitespace, nor plus or minus signs.
+
+    :param text: The string to convert
+    :raises ValueError: Raises if the string has any non-numeric characters
+    :return: [description]
+    :rtype: [type]
+    """
+    if text.strip("+-\t ") != text:
+        raise ValueError(r"invalid literal for int() with base 10: {text}")
+    return int(text)
+
+
 def _parse_linter_line(line: str, root: Path) -> Tuple[Path, int, str, str]:
-    # Parse an error/note line.
-    # Given: line == "dir/file.py:123: error: Foo\n"
-    # Sets: path = Path("abs/path/to/dir/file.py:123"
-    #       linenum = 123
-    #       description = "error: Foo"
+    """Parse one line of linter output
+
+    Only parses lines with
+    - a file path (without leading-trailing whitespace),
+    - a non-negative line number (without leading/trailing whitespace),
+    - optionally a column number (without leading/trailing whitespace), and
+    - a description.
+
+    Examples of successfully parsed lines::
+
+        path/to/file.py:42: Description
+        path/to/file.py:42:5: Description
+
+    Given a root of ``Path("path/")``, these would be parsed into::
+
+        (Path("to/file.py"), 42, "path/to/file.py:42:", "Description")
+        (Path("to/file.py"), 42, "path/to/file.py:42:5:", "Description")
+
+    For all other lines, a dummy entry is returned: an empty path, zero as the line
+    number, an empty location string and an empty description. Such lines should be
+    simply ignored, since many linters display supplementary information insterspersed
+    with the actual linting notifications.
+
+    :param line: The linter output line to parse. May have a trailing newline.
+    :param root: The root directory to resolve full file paths against
+    :return: A 4-tuple of
+             - a ``root``-relative file path,
+             - the line number,
+             - the path and location string, and
+             - the description.
+
+    """
     try:
-        location, description = line[:-1].split(": ", 1)
+        location, description = line.rstrip().split(": ", 1)
         if location[1:3] == ":\\":
             # Absolute Windows paths need special handling. Separate out the ``C:`` (or
             # similar), then split by colons, and finally re-insert the ``C:``.
@@ -46,12 +88,14 @@ def _parse_linter_line(line: str, root: Path) -> Tuple[Path, int, str, str]:
             path_str = f"{location[:2]}{path_in_drive}"
         else:
             path_str, linenum_str, *rest = location.split(":")
-        linenum = int(linenum_str)
+        if path_str.strip() != path_str:
+            raise ValueError(r"Filename {path_str!r} has leading/trailing whitespace")
+        linenum = _strict_nonneg_int(linenum_str)
         if len(rest) > 1:
-            raise ValueError("Too many colon-separated tokens")
+            raise ValueError("Too many colon-separated tokens in {location!r}")
         if len(rest) == 1:
             # Make sure it column looks like an int on "<path>:<linenum>:<column>"
-            _column = int(rest[0])  # noqa: F841
+            _column = _strict_nonneg_int(rest[0])  # noqa: F841
     except ValueError:
         # Encountered a non-parsable line which doesn't express a linting error.
         # For example, on Mypy:

--- a/src/darker/tests/conftest.py
+++ b/src/darker/tests/conftest.py
@@ -1,9 +1,10 @@
 """Configuration and fixtures for the Pytest based test suite"""
 
 import os
+import re
 from pathlib import Path
 from subprocess import check_call  # nosec
-from typing import Dict, Union
+from typing import Dict, Iterable, List, Union
 
 import pytest
 from black import find_project_root as black_find_project_root
@@ -83,6 +84,21 @@ class GitRepoFixture:
     def create_branch(self, new_branch: str, start_point: str) -> None:
         """Fixture method to create and check out new branch at given starting point"""
         self._run("checkout", "-b", new_branch, start_point)
+
+    def expand_root(self, lines: Iterable[str]) -> List[str]:
+        """Replace "{root/<path>}" in strings with the path in the temporary Git repo
+
+        This is used to generate expected strings corresponding to locations of files in
+        the temporary Git repository.
+
+        :param lines: The lines of text to process
+        :return: Given lines with paths processed
+
+        """
+        return [
+            re.sub(r"\{root/(.*?)\}", lambda m: str(self.root / str(m.group(1))), line)
+            for line in lines
+        ]
 
 
 @pytest.fixture

--- a/src/darker/tests/test_linting.py
+++ b/src/darker/tests/test_linting.py
@@ -28,25 +28,36 @@ from darker.tests.helpers import raises_if_exception
     dict(line="mod.py:invalid-linenum:5: Description\n", expect=(Path(), 0, "", "")),
     dict(line="invalid linter output\n", expect=(Path(), 0, "", "")),
     dict(
-        line="{git_repo.root}{sep}mod.py:42: Full path\n",
-        expect=(Path("mod.py"), 42, "{git_repo.root}{sep}mod.py:42:", "Full path"),
+        line="{git_root_absolute}{sep}mod.py:42: Full path\n",
+        expect=(
+            Path("mod.py"),
+            42,
+            "{git_root_absolute}{sep}mod.py:42:",
+            "Full path",
+        ),
     ),
     dict(
-        line="{git_repo.root}{sep}mod.py:42:5: Full path\n",
-        expect=(Path("mod.py"), 42, "{git_repo.root}{sep}mod.py:42:5:", "Full path"),
+        line="{git_root_absolute}{sep}mod.py:42:5: Full path\n",
+        expect=(
+            Path("mod.py"),
+            42,
+            "{git_root_absolute}{sep}mod.py:42:5:",
+            "Full path",
+        ),
     ),
 )
 def test_parse_linter_line(git_repo, monkeypatch, line, expect):
     """Linter output is parsed correctly"""
     monkeypatch.chdir(git_repo.root)
-    line_expanded = line.format(git_repo=git_repo, sep=os.sep)
+    root_abs = git_repo.root.absolute()
+    line_expanded = line.format(git_root_absolute=root_abs, sep=os.sep)
 
     result = linting._parse_linter_line(line_expanded, git_repo.root)
 
     expect_expanded = (
         expect[0],
         expect[1],
-        expect[2].format(git_repo=git_repo, sep=os.sep),
+        expect[2].format(git_root_absolute=root_abs, sep=os.sep),
         expect[3],
     )
     assert result == expect_expanded

--- a/src/darker/tests/test_linting.py
+++ b/src/darker/tests/test_linting.py
@@ -17,16 +17,13 @@ from darker.tests.helpers import raises_if_exception
 
 @pytest.mark.kwparametrize(
     dict(
-        line="module.py:42: Description\n",
-        expect=(Path("module.py"), 42, "module.py:42:", "Description"),
+        line="module.py:42: Just a line number\n",
+        expect=(Path("module.py"), 42, "module.py:42:", "Just a line number"),
     ),
     dict(
-        line="module.py:42:5: Description\n",
-        expect=(Path("module.py"), 42, "module.py:42:5:", "Description"),
+        line="module.py:42:5: With column  \n",
+        expect=(Path("module.py"), 42, "module.py:42:5:", "With column"),
     ),
-    dict(line="no-linenum.py: Description\n", expect=(Path(), 0, "", "")),
-    dict(line="mod.py:invalid-linenum:5: Description\n", expect=(Path(), 0, "", "")),
-    dict(line="invalid linter output\n", expect=(Path(), 0, "", "")),
     dict(
         line="{git_root_absolute}{sep}mod.py:42: Full path\n",
         expect=(
@@ -37,14 +34,39 @@ from darker.tests.helpers import raises_if_exception
         ),
     ),
     dict(
-        line="{git_root_absolute}{sep}mod.py:42:5: Full path\n",
+        line="{git_root_absolute}{sep}mod.py:42:5: Full path with column\n",
         expect=(
             Path("mod.py"),
             42,
             "{git_root_absolute}{sep}mod.py:42:5:",
-            "Full path",
+            "Full path with column",
         ),
     ),
+    dict(
+        line="mod.py:42: 123 digits start the description\n",
+        expect=(Path("mod.py"), 42, "mod.py:42:", "123 digits start the description"),
+    ),
+    dict(
+        line="mod.py:42:    indented description\n",
+        expect=(Path("mod.py"), 42, "mod.py:42:", "   indented description"),
+    ),
+    dict(
+        line="mod.py:42:5:    indented description\n",
+        expect=(Path("mod.py"), 42, "mod.py:42:5:", "   indented description"),
+    ),
+    dict(line="mod.py: No line number\n", expect=(Path(), 0, "", "")),
+    dict(line="mod.py:foo:5: Invalid line number\n", expect=(Path(), 0, "", "")),
+    dict(line="mod.py:42:bar: Invalid column\n", expect=(Path(), 0, "", "")),
+    dict(line="invalid linter output\n", expect=(Path(), 0, "", "")),
+    dict(line=" leading:42: whitespace\n", expect=(Path(), 0, "", "")),
+    dict(line=" leading:42:5 whitespace and column\n", expect=(Path(), 0, "", "")),
+    dict(line="trailing :42: filepath whitespace\n", expect=(Path(), 0, "", "")),
+    dict(line="leading: 42: linenum whitespace\n", expect=(Path(), 0, "", "")),
+    dict(line="trailing:42 : linenum whitespace\n", expect=(Path(), 0, "", "")),
+    dict(line="plus:+42: before linenum\n", expect=(Path(), 0, "", "")),
+    dict(line="minus:-42: before linenum\n", expect=(Path(), 0, "", "")),
+    dict(line="plus:42:+5 before column\n", expect=(Path(), 0, "", "")),
+    dict(line="minus:42:-5 before column\n", expect=(Path(), 0, "", "")),
 )
 def test_parse_linter_line(git_repo, monkeypatch, line, expect):
     """Linter output is parsed correctly"""

--- a/src/darker/tests/test_linting.py
+++ b/src/darker/tests/test_linting.py
@@ -2,6 +2,7 @@
 
 """Unit tests for :mod:`darker.linting`"""
 
+import os
 import re
 from pathlib import Path
 from textwrap import dedent
@@ -27,25 +28,25 @@ from darker.tests.helpers import raises_if_exception
     dict(line="mod.py:invalid-linenum:5: Description\n", expect=(Path(), 0, "", "")),
     dict(line="invalid linter output\n", expect=(Path(), 0, "", "")),
     dict(
-        line="{git_repo.root}/mod.py:42: Full path\n",
-        expect=(Path("mod.py"), 42, "{git_repo.root}/mod.py:42:", "Full path"),
+        line="{git_repo.root}{sep}mod.py:42: Full path\n",
+        expect=(Path("mod.py"), 42, "{git_repo.root}{sep}mod.py:42:", "Full path"),
     ),
     dict(
-        line="{git_repo.root}/mod.py:42:5: Full path\n",
-        expect=(Path("mod.py"), 42, "{git_repo.root}/mod.py:42:5:", "Full path"),
+        line="{git_repo.root}{sep}mod.py:42:5: Full path\n",
+        expect=(Path("mod.py"), 42, "{git_repo.root}{sep}mod.py:42:5:", "Full path"),
     ),
 )
 def test_parse_linter_line(git_repo, monkeypatch, line, expect):
     """Linter output is parsed correctly"""
     monkeypatch.chdir(git_repo.root)
-    line_expanded = line.format(git_repo=git_repo)
+    line_expanded = line.format(git_repo=git_repo, sep=os.sep)
 
     result = linting._parse_linter_line(line_expanded, git_repo.root)
 
     expect_expanded = (
         expect[0],
         expect[1],
-        expect[2].format(git_repo=git_repo),
+        expect[2].format(git_repo=git_repo, sep=os.sep),
         expect[3],
     )
     assert result == expect_expanded

--- a/src/darker/tests/test_linting.py
+++ b/src/darker/tests/test_linting.py
@@ -3,6 +3,7 @@
 """Unit tests for :mod:`darker.linting`"""
 
 import os
+import sys
 from pathlib import Path
 from textwrap import dedent
 from unittest.mock import call, patch
@@ -12,6 +13,9 @@ import pytest
 from darker import linting
 from darker.git import WORKTREE, RevisionRange
 from darker.tests.helpers import raises_if_exception
+
+SKIP_ON_WINDOWS = [pytest.mark.skip] if sys.platform.startswith("win") else []
+SKIP_ON_UNIX = [] if sys.platform.startswith("win") else [pytest.mark.skip]
 
 
 @pytest.mark.kwparametrize(
@@ -194,7 +198,7 @@ def test_check_linter_output():
         ],
     ),
     dict(
-        _descr="Message for a file outside the common root is ignored with a warning",
+        _descr="Message for file outside common root is ignored with a warning (Unix)",
         paths=["one.py"],
         location="/elsewhere/mod.py:1:",
         expect_output=[],
@@ -202,6 +206,18 @@ def test_check_linter_output():
             "WARNING Linter message for a file /elsewhere/mod.py "
             "outside requested directory {root/}"
         ],
+        marks=SKIP_ON_WINDOWS,
+    ),
+    dict(
+        _descr="Message for file outside common root is ignored with a warning (Win)",
+        paths=["one.py"],
+        location="C:\\elsewhere\\mod.py:1:",
+        expect_output=[],
+        expect_log=[
+            "WARNING Linter message for a file C:\\elsewhere\\mod.py "
+            "outside requested directory {root/}"
+        ],
+        marks=SKIP_ON_UNIX,
     ),
 )
 def test_run_linter(

--- a/src/darker/tests/test_linting.py
+++ b/src/darker/tests/test_linting.py
@@ -3,7 +3,6 @@
 """Unit tests for :mod:`darker.linting`"""
 
 import os
-import re
 from pathlib import Path
 from textwrap import dedent
 from unittest.mock import call, patch
@@ -54,9 +53,14 @@ from darker.tests.helpers import raises_if_exception
         line="mod.py:42:5:    indented description\n",
         expect=(Path("mod.py"), 42, "mod.py:42:5:", "   indented description"),
     ),
+    dict(
+        line="nonpython.txt:5: Non-Python file\n",
+        expect=(Path("nonpython.txt"), 5, "nonpython.txt:5:", "Non-Python file"),
+    ),
     dict(line="mod.py: No line number\n", expect=(Path(), 0, "", "")),
     dict(line="mod.py:foo:5: Invalid line number\n", expect=(Path(), 0, "", "")),
     dict(line="mod.py:42:bar: Invalid column\n", expect=(Path(), 0, "", "")),
+    dict(line="/outside/mod.py:5: Outside the repo\n", expect=(Path(), 0, "", "")),
     dict(line="invalid linter output\n", expect=(Path(), 0, "", "")),
     dict(line=" leading:42: whitespace\n", expect=(Path(), 0, "", "")),
     dict(line=" leading:42:5 whitespace and column\n", expect=(Path(), 0, "", "")),
@@ -179,6 +183,26 @@ def test_check_linter_output():
         expect_output=[],
         expect_log=["WARNING Missing file missing.py from echo missing.py:1:"],
     ),
+    dict(
+        _descr="Linter message for a non-Python file is ignored with a warning",
+        paths=["one.py"],
+        location="nonpython.txt:1:",
+        expect_output=[],
+        expect_log=[
+            "WARNING Linter message for a non-Python file: "
+            "nonpython.txt:1: {root/one.py}"
+        ],
+    ),
+    dict(
+        _descr="Message for a file outside the common root is ignored with a warning",
+        paths=["one.py"],
+        location="/elsewhere/mod.py:1:",
+        expect_output=[],
+        expect_log=[
+            "WARNING Linter message for a file /elsewhere/mod.py "
+            "outside requested directory {root/}"
+        ],
+    ),
 )
 def test_run_linter(
     git_repo, capsys, caplog, _descr, paths, location, expect_output, expect_log
@@ -196,7 +220,9 @@ def test_run_linter(
           test.py:1: git-repo-root/one.py git-repo-root/two.py
 
     """
-    src_paths = git_repo.add({"test.py": "1\n2\n"}, commit="Initial commit")
+    src_paths = git_repo.add(
+        {"test.py": "1\n2\n", "nonpython.txt": "hello\n"}, commit="Initial commit"
+    )
     src_paths["test.py"].write_bytes(b"one\n2\n")
     cmdline = f"echo {location}"
     revrange = RevisionRange("HEAD", ":WORKTREE:")
@@ -209,12 +235,9 @@ def test_run_linter(
     # by checking standard output from the our `echo` "linter".
     # The test cases also verify that only linter reports on modified lines are output.
     result = capsys.readouterr().out.splitlines()
-    assert result == [
-        re.sub(r"\{root/(.*?)\}", lambda m: str(git_repo.root / m.group(1)), line)
-        for line in expect_output
-    ]
+    assert result == git_repo.expand_root(expect_output)
     logs = [f"{record.levelname} {record.message}" for record in caplog.records]
-    assert logs == expect_log
+    assert logs == git_repo.expand_root(expect_log)
 
 
 def test_run_linter_non_worktree():

--- a/src/darker/tests/test_linting.py
+++ b/src/darker/tests/test_linting.py
@@ -26,14 +26,29 @@ from darker.tests.helpers import raises_if_exception
     dict(line="no-linenum.py: Description\n", expect=(Path(), 0, "", "")),
     dict(line="mod.py:invalid-linenum:5: Description\n", expect=(Path(), 0, "", "")),
     dict(line="invalid linter output\n", expect=(Path(), 0, "", "")),
+    dict(
+        line="{git_repo.root}/mod.py:42: Full path\n",
+        expect=(Path("mod.py"), 42, "{git_repo.root}/mod.py:42:", "Full path"),
+    ),
+    dict(
+        line="{git_repo.root}/mod.py:42:5: Full path\n",
+        expect=(Path("mod.py"), 42, "{git_repo.root}/mod.py:42:5:", "Full path"),
+    ),
 )
 def test_parse_linter_line(git_repo, monkeypatch, line, expect):
     """Linter output is parsed correctly"""
     monkeypatch.chdir(git_repo.root)
+    line_expanded = line.format(git_repo=git_repo)
 
-    result = linting._parse_linter_line(line, git_repo.root)
+    result = linting._parse_linter_line(line_expanded, git_repo.root)
 
-    assert result == expect
+    expect_expanded = (
+        expect[0],
+        expect[1],
+        expect[2].format(git_repo=git_repo),
+        expect[3],
+    )
+    assert result == expect_expanded
 
 
 @pytest.mark.kwparametrize(

